dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -75,13 +69,114 @@ trait Udf {
     * apache/spark
     */
   def udf[T, A1, A2, A3, A4, A5, R: TypedEncoder](f: (A1,A2,A3,A4,A5) => R):
-    (TypedColumn[T, A1], TypedColumn[T, A2], TypedColumn[T, A3], TypedColumn[T, A4],  TypedColumn[T, A5]) => TypedColumn[T, R] = {
-    (u1, u2, u3, u4, u5) =>
-      val (a1enc, a2enc, a3enc, a4enc, a5enc) = (u1.uencoder, u2.uencoder, u3.uencoder, u4.uencoder, u5.uencoder)
-
-      val scalaUdf = ScalaUDF(f, TypedEncoder[R].targetDataType, Seq(u1.expr, u2.expr, u3.expr, u4.expr, u5.expr),
-        Seq(a1enc.targetDataType, a2enc.targetDataType, a3enc.targetDataType, a4enc.targetDataType, a5enc.targetDataType))
+    (TypedColumn[T, A1], TypedColumn[T, A2], TypedColumn[T, A3], TypedColumn[T, A4], TypedColumn[T, A5]) => TypedColumn[T, R] = {
+    case us =>
+      val scalaUdf = FramelessUdf(f, us.toList[UntypedExpression[T]], TypedEncoder[R])
       new TypedColumn[T, R](scalaUdf)
     }
 }
 
+case class FramelessUdf[T, R](
```

Comment:
I don't think we should stop people from accessing internals if they want, what if we document that this is internal?